tests/unit/python/foglamp/services/core/api/test_configuration.py

```diff
@@ -641,3 +641,50 @@ def async_mock():
                 patch_cat_all_item.assert_called_once_with(category_name=name)
             patch_create_cat.assert_called_once_with(category_name=name, category_description=desc,
                                                      category_value=info, keep_original_items=False, display_name=None)
+
+    async def test_update_bulk_config_bad_request(self, client, category_name='rest_api'):
+        resp = await client.put('/foglamp/category/{}'.format(category_name), data=json.dumps({}))
+        assert 400 == resp.status
+        assert 'Nothing to update' == resp.reason
+
+    @pytest.mark.parametrize("code, exception_name", [
+        (404, [NameError, KeyError]),
+        (400, [ValueError, TypeError]),
+        (500, Exception)
+    ])
+    async def test_update_bulk_config_exception(self, client,  code, exception_name, category_name='rest_api'):
+        payload = {"http_port": "8082", "authentication": "mandatory"}
+        storage_client_mock = MagicMock(spec=StorageClientAsync)
+        c_mgr = ConfigurationManager(storage_client_mock)
+        with patch.object(connect, 'get_storage_async', return_value=storage_client_mock):
+            with patch.object(c_mgr, 'update_configuration_item_bulk', side_effect=exception_name) as patch_update_bulk:
+                resp = await client.put('/foglamp/category/{}'.format(category_name), data=json.dumps(payload))
+                assert code == resp.status
+                assert resp.reason is None
+            patch_update_bulk.assert_called_once_with(category_name, payload)
+
+    @pytest.mark.parametrize("category_name", [
+        "rest_api", "Rest $API"
```

Comment:
@pytest.mark.parametrize("category_name", [
        "rest_api", "Rest $API"

    @pytest.mark.parametrize("encoded_category_name", "category_name", [
        "rest_api", "Rest%20%24API", ["rest_api", "Rest $API"])

Reply:
Okay, This is not needed. 
> aiohttp internally performs URL canonization before sending request.